extra_data/tests/test_sourcedata.py

```diff
@@ -140,6 +140,21 @@ def test_device_class(mock_spb_raw_run):
     assert xgm_inst.device_class is None
 
 
+def test_path_traits(mock_spb_raw_run):
+    run = RunDirectory(mock_spb_raw_run)
+    xgm = run['SPB_XTD9_XGM/DOOCS/MAIN']
+
+    assert xgm.storage_class is None  # Not an EuXFEL path.
+    assert xgm.data_category == 'RAW'
+    assert xgm.aggregator == 'DA01'
+
+    run = RunDirectory(mock_spb_raw_run).select_trains(np.s_[:0])
+    xgm = run['SPB_XTD9_XGM/DOOCS/MAIN']
+
+    with pytest.raises(NoDataError):
```

Comment:
Looks like the code is still raising ValueError.

Reply:
Yes, I'm sorry for the commit spam, I don't seem to be fully concentrated. I happened to be in `tests/` and didn't commit the changes to `../sourcedata.py`.